internal/daemon/archive_http_test.go

```
@@ -0,0 +1,294 @@
+
+	sess := decodeJSON[sessionAPIResponse](t, doRequest(t, d, "POST", "/sessions", createSessionRequest{
+		Name:         "disposition-test",
+		WorkspaceDir: ws,
+	}))
+
+	writeTestArchive(t, ws, sess.ID)
+
+	server := httptest.NewServer(d.server.Handler)
+	defer server.Close()
+
+	resp, err := http.Get(server.URL + "/sessions/" + sess.ID + "/archive.tar.gz")
```

golangci-lint (2.11.4)

[error] net/http.Get must not be called. use net/http.NewRequestWithContext and (*net/http.Client).Do(*http.Request)

(noctx)